Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentAdapter.h

```diff
@@ -182,11 +199,15 @@ namespace AZ::DocumentPropertyEditor
         //! This patch should apply cleanly on the last result GetContents would have returned after any preceding changed
         //! or reset events.
         void NotifyContentsChanged(const Dom::Patch& patch);
+        //! Subclasses may call this to trigger a PropertyChangeEvent to notify the view that one of this adapter's
+        //! property editor instances has altered its value.
+        void NotifyPropertyChanged(const PropertyChangeInfo& changeInfo);
```

Comment:
Hm I was thinking of having this notification specifically for `ReflectionAdapter` since it only really makes sense in that context - otherwise every adapter sort of implicitly has to implement this, when really it's property changes in the reflection system that we want to hook.

This might sound a little weird/pedantic, but it's a design thing: the aspiration is that business logic like "when this property changes, do X" is handled not by the view but by the adapter, to make complicated side effects like that encapsulated from the presentation layer and easier to test.

Reply:
I don't think it's pedantic at all. My understanding of the DPE is still growing, so it's more of a misunderstanding on my part. I was thinking all DocumentAdapters were for property editing applications and not thinking of the ReflectionAdapter as being _the_ adapter for property editors (which makes sense because these properties are reflected). I see how the DocumentAdapter shouldn't necessarily need knowledge of the reflection anything (should have been a clue when all I ended up doing in DocumentAdapter::NotifyPropertyChanged is forwarding those event params along).

I'll get that changed!